### PR TITLE
Site admin only can delete site and add/remove site collaborators

### PR DESCRIPTION
**Pull request by amirbey**

### CODE REVIEW

File: api/controllers/site.js [hunk 1]
```diff
@@ -68,7 +68,7 @@ module.exports = {
         siteJSON = json;
         return authorizer.destroy(req.user, site);
       })
-      .then(() => S3SiteRemover.removeSite(site))
+      .then(() => S3SiteRemover.removeSite(site, req.user))
```
Comment (jmhooper) [inline api/controllers/site.js]: What are your thoughts on moving the GH check to the authorizer? That way the `S3SiteRemove` stays focused on removing sites, and the policies for whether someone is authorized to remove a site are colocated.

**Comment by wslack** [comment]
Amir fixing some tests

**Comment by amirbey** [comment]
@jmhooper - thank for the suggestion. changes made .. pls re-review

**Comment by amirbey** [comment]
@jmhooper - pls re-review

**Pull request merged**